### ISSUE TITLE
Bug: Imports do not seems to be handled correctly in 0.720

### ISSUE THREAD

**Issue by futursolo** [python/mypy#7247]
I recently bumped my Mypy version from 0.701 to 0.720. It seems like Mypy's starting to process imported modules residing in the same directory as the one that's being checked. But it does not handle them correctly.

Environment:
```
macOS 10.14.5
Python 3.7.4
Mypy 0.720/0.701
```

Directory Structure:
```
|
|-- mod_a
|  |-- __init__.py
|
|-- mod_b
   |-- __init__.py
   |-- b.py
```

Content of `mod_a/__init__.py`:
```python
import mod_b

class A(mod_b.B):
    pass
```

Content of `mod_b/__init__.py`:
```python
from .b import B
```

Content of `mod_b/b.py`:
```python
class B:
    pass
```

Command:
```shell
mypy --python-version 3.7 \
    --strict --show-traceback \
    --ignore-missing-imports \
    --show-error-context \
    -p mod_a
```

Error message in 0.720(no error in 0.701):
```
mod_a/__init__.py:4: error: Name 'mod_b.B' is not defined
mod_a/__init__.py: note: In class "A":
mod_a/__init__.py:4: error: Class cannot subclass 'B' (has type 'Any')
```

**Comment by JelleZijlstra** [comment]
This is the expected behavior of the new `--no-implicit-reexport` flag, which is included in `--strict`. To make your code pass, use `from .b import B as B` in `mod_b/__init__.py`.

This is documented in https://mypy.readthedocs.io/en/latest/command_line.html#miscellaneous-strictness-flags.

**Comment by futursolo** [comment]
Is there any chance we can make the error message clearer?

**Comment by JukkaL** [comment]
Yes, the error message should be clearer. Created #7251 to track this.

**Comment by futursolo** [comment]
Also, upon further testing, `--no-implicit-reexport` does not seem to respect `__all__`either. Shouldn't variable names listed in `__all__` also be considered as exported names?

**Comment by JelleZijlstra** [comment]
That's being fixed in #7099.
